internal/app/secretless/providers/conjur/provider.go

```diff
@@ -105,8 +102,15 @@ func ProviderFactory(options plugin_v1.ProviderOptions) (plugin_v1.Provider, err
 		}
 		provider.Authenticator = authenticator
 
-		// Kick off the goroutine that will maintain the Conjur access token
-		go provider.refreshAccessToken()
+		refreshErr := provider.refreshAccessToken(true)
+		if refreshErr != nil {
+			return nil, refreshErr
+		}
+
+		go func() {
+			time.Sleep(provider.Authenticator.Config.TokenRefreshTimeout)
+			provider.refreshAccessToken(false)
```

Comment:
Do we want to also exit if the refresh fails here? What will happen if we don't, it'll try again in 6 mins?

Reply:
We can't really "exit" the creation of the provider from an async goroutine - maybe we need to think about if we want to panic out (which is doable here) if the refresh fails?

Reply:
per convo w/ @doodlesbykumbi we'll defer this since the main loop will kill the pod on next secrets fetch. We can figure out what to do with this in another iteration if needed.